loki.source.docker: deduplicate matching container IDs (#6055)
grafana/agent: docs/sources/flow/reference/components/loki.source.docker.md

stores the read offsets so that if there is a component or Agent restart,
`loki.source.docker` can pick up tailing from the same spot.

> In case the targets argument contains multiple entries with the same container

Reviewer: ```suggestion
If the target's argument contains multiple entries with the same container
```

> ID (for example as a result of `discovery.docker` picking up multiple exposed
> ports or networks) `loki.source.docker` will deduplicate them, and only keep

Reviewer: ```suggestion
ID (for example as a result of `discovery.docker` picking up multiple exposed
ports or networks), `loki.source.docker` will deduplicate them, and only keep
```

Generally, we should avoid using brackets in the text. I'm leaving this one in for now because removing it makes for a VERY long sentence (or forces a big rewrite).

the first of each container ID instances, based on the

> `__meta_docker_container_id` label.  As such, the Docker daemon will be queried

Reviewer: ```suggestion
`__meta_docker_container_id` label.  As such, the Docker daemon is queried
```
